pydantic_ai_slim/pydantic_ai/_utils.py

```diff
@@ -41,8 +52,33 @@
 _P = ParamSpec('_P')
 _R = TypeVar('_R')
 
+_prefer_blocking_execution: ContextVar[bool] = ContextVar('_prefer_blocking_execution', default=False)
+
+
+@contextmanager
+def blocking_execution() -> Iterator[None]:
```

Comment:
`with _utils.blocking_execution()` makes it seems like all execution will be blocking, but that's not true because of async etc. So maybe we can rename this to something like `with disable_threads()`?